core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooLock.java

```diff
@@ -139,26 +144,66 @@ public synchronized boolean tryLock(LockWatcher lw, byte[] data)
     return false;
   }
 
+  private static void sortChildrenByLockPrefix(List<String> children) {
+    Collections.sort(children, new Comparator<String>() {
+      @Override
+      public int compare(String o1, String o2) {
+
+        // Lock should be of the form:
+        // zlock#UUID#sequenceNumber
+        // Example:
+        // zlock#44755fbe-1c9e-40b3-8458-03abaf950d7e#0000000000
+
+        boolean lValid = true;
+        if (o1.length() <= 43) {
+          lValid = false;
+        }
+        boolean rValid = true;
+        if (o2.length() <= 43) {
+          rValid = false;
+        }
+        // Make any invalid's sort last (higher)
+        if (lValid && rValid) {
+          int leftIdx = o1.lastIndexOf('#');
+          int rightIdx = o2.lastIndexOf('#');
+          return Integer.compare(Integer.parseInt(o1.substring(leftIdx)),
+              Integer.parseInt(o2.substring(rightIdx)));
+        } else if (!lValid && rValid) {
```

Comment:
This should be able to be a string compare - zookeeper pads the sequence number with 0's (from zookeerer docs: 
`The counter has a format of %010d -- that is 10 digits with 0 (zero) padding`